Framework/DataHandling/src/LoadBankFromDiskTask.cpp

```diff
@@ -0,0 +1,909 @@
+             "entry.\n";
+      // This'll make the code skip looking for any pulse times.
+      pulse_i = numPulses + 1;
+    }
+
+    prog->report(entry_name + ": filling events");
+
+    // Will we need to compress?
+    bool compress = (alg->compressTolerance >= 0);
+
+    // Which detector IDs were touched? - only matters if compress is on
+    std::vector<bool> usedDetIds;
+    if (compress)
+      usedDetIds.assign(m_max_id - m_min_id + 1, false);
+
+    // Go through all events in the list
+    for (std::size_t i = 0; i < numEvents; i++) {
+      //------ Find the pulse time for this event index ---------
+      if (pulse_i < numPulses - 1) {
+        bool breakOut = false;
+        // Go through event_index until you find where the index increases to
+        // encompass the current index. Your pulse = the one before.
+        while ((i + startAt < event_index->operator[](pulse_i)) ||
+               (i + startAt >= event_index->operator[](pulse_i + 1))) {
+          pulse_i++;
+          // Check once every new pulse if you need to cancel (checking on every
+          // event might slow things down more)
+          if (alg->getCancel())
+            breakOut = true;
+          if (pulse_i >= (numPulses - 1))
+            break;
+        }
+
+        // Save the pulse time at this index for creating those events
+        pulsetime = thisBankPulseTimes->pulseTimes[pulse_i];
+        int logPeriodNumber = thisBankPulseTimes->periodNumbers[pulse_i];
+        periodNumber = logPeriodNumber > 0
+                           ? logPeriodNumber
+                           : periodNumber; // Some historic files have recorded
+                                           // their logperiod numbers as zeros!
+        periodIndex = periodNumber - 1;
+
+        // Determine if pulse times continue to increase
+        if (pulsetime < lastpulsetime)
+          pulsetimesincreasing = false;
+        else
+          lastpulsetime = pulsetime;
+
+        // Flag to break out of the event loop without using goto
+        if (breakOut)
+          break;
+      }
+
+      // We cached a pointer to the vector<tofEvent> -> so retrieve it and add
+      // the event
+      detid_t detId = event_id[i];
+      if (detId >= m_min_id && detId <= m_max_id) {
+        // Create the tofevent
+        double tof = static_cast<double>(event_time_of_flight[i]);
+        if ((tof >= alg->filter_tof_min) && (tof <= alg->filter_tof_max)) {
+          // Handle simulated data if present
+          if (have_weight) {
+            double weight = static_cast<double>(event_weight[i]);
+            double errorSq = weight * weight;
+            Mantid::DataHandling::LoadEventNexus::WeightedEventVector_pt
+                eventVector = alg->weightedEventVectors[periodIndex][detId];
+            // NULL eventVector indicates a bad spectrum lookup
+            if (eventVector) {
+#if !(defined(__INTEL_COMPILER)) && !(defined(__clang__))
```

Comment:
Would you also remove both this macro and the very similar one on line 219? `emplace_back` is available on all of our supported platforms.